File: mobile-app/ios/rnapp/WasmModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(WasmModule, NSObject)

RCT_EXTERN_METHOD(add:(NSInteger)a
                  b:(NSInteger)b
                  resolver:(RCTPromiseResolveBlock)resolver
                  rejecter:(RCTPromiseRejectBlock)rejecter)

@end

File: mobile-app/ios/rnapp/WasmModule.swift
import Foundation
import React

@objc(WasmModule)
class WasmModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func add(_ a: Int, b: Int, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
    do {
      let result = WasmManager.shared.callAdd(a: a, b: b)
      resolver(result)
    } catch {
      rejecter("WASM_ERROR", "Failed to execute WASM add function", error)
    }
  }
}
